eclair-front/src/main/scala/fr/acinq/eclair/FrontSetup.scala

```diff
@@ -0,0 +1,108 @@
+import fr.acinq.bitcoin.Crypto.{PrivateKey, PublicKey}
+import fr.acinq.eclair.crypto.Noise.KeyPair
+import fr.acinq.eclair.io.Switchboard.{GetRouterPeerConf, RouterPeerConf}
+import fr.acinq.eclair.io.{ClientSpawner, Server}
+import fr.acinq.eclair.router.FrontRouter
+import grizzled.slf4j.Logging
+import scodec.bits.ByteVector
+
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future, Promise}
+
+class FrontSetup(datadir: File)(implicit system: ActorSystem) extends Logging {
+
+  implicit val timeout = Timeout(30 seconds)
+  implicit val ec: ExecutionContext = system.dispatcher
+
+  logger.info(s"hello!")
+  logger.info(s"version=${getClass.getPackage.getImplementationVersion} commit=${getClass.getPackage.getSpecificationVersion}")
+  logger.info(s"datadir=${datadir.getCanonicalPath}")
+  logger.info(s"initializing secure random generator")
+  // this will force the secure random instance to initialize itself right now, making sure it doesn't hang later (see comment in package.scala)
+  secureRandom.nextInt()
+
+  datadir.mkdirs()
+
+  val config = system.settings.config.getConfig("eclair")
+
+  val keyPair = {
+    val pub = ByteVector.fromValidHex(config.getString("front.pub"))
+    val priv = config.getString("front.priv-key-provider") match {
+      case "aws-sm" =>
+        val sm = AWSSecretsManagerClient.builder().build()
+        try {
+          // we retrieve the node key from AWS secrets manager and we compare the corresponding pub key with the expected one
+          val secretId = config.getString("front.aws-sm.priv-key-name")
+          ByteVector.fromValidHex(sm.getSecretValue(new GetSecretValueRequest().withSecretId(secretId)).getSecretString)
+        } finally {
+          sm.shutdown()
+        }
+      case "env" => ByteVector.fromValidHex(config.getString("front.priv-key"))
+    }
+    val keyPair = KeyPair(pub, priv)
+    require(PrivateKey(priv).publicKey == PublicKey(pub), "priv/pub keys mismatch")
+    keyPair
+  }
+
+  logger.info(s"nodeid=${keyPair.pub.toHex}")
+
+  val serverBindingAddress = new InetSocketAddress(
+    config.getString("server.binding-ip"),
+    config.getInt("server.port"))
+
+  def bootstrap: Future[Unit] = {
+
+    val backendAddressFound = Promise[Address]()
+    val tcpBound = Promise[Done]()
+
+    for {
+      _ <- Future.successful(0)
+
+      _ = system.actorOf(Props(new ClusterListener(backendAddressFound)), name = "cluster-listener")
+      backendAddress <- backendAddressFound.future
+
+      switchBoardSelection = system.actorSelection(RootActorPath(backendAddress) / "user" / "*" / "switchboard")
+      remoteSwitchboard <- switchBoardSelection.resolveOne()
+      routerSelection = system.actorSelection(RootActorPath(backendAddress) / "user" / "*" / "router")
+      remoteRouter <- routerSelection.resolveOne()
+
+      RouterPeerConf(routerConf, peerConnectionConf) <- (switchBoardSelection ? GetRouterPeerConf).mapTo[RouterPeerConf]
```

Comment:
Can't we do:

```suggestion
      RouterPeerConf(routerConf, peerConnectionConf) <- (remoteSwitchboard ? GetRouterPeerConf).mapTo[RouterPeerConf]
```

Reply:
Yes, and it's probably easier to understand.